api/v1beta1/network_types.go

```diff
@@ -174,10 +174,28 @@ type VPCSpec struct {
 	// Defaults to 10.0.0.0/16.
 	CidrBlock string `json:"cidrBlock,omitempty"`
 
+	// EnableIPv6 requests an Amazon-provided IPv6 CIDR block with a /56 prefix length for
+	// the VPC. You cannot specify the range of IP addresses, or the size of the
+	// CIDR block.
+	// +optional
+	EnableIPv6 bool `json:"enableIPv6"`
```

Comment:
I'm wondering if we group  `EnableIPv6`, `IPv6CidrBlock`, `IPv6Pool` within an IPv6 specific struct that we use here?

So you end with something like this (just an illustration though):

```yaml
ipv6:
  enable: true
  cidrBlock: "XXXXXX"
  pool: "XXXXX"
```

Reply:
Good point! I think that would be cool. :)

Reply:
Uh that's quite a lot of moving around things. :D But it makes sense to group them. :)

Reply:
Hmmmm... It makes some code a bit unwildely. Like, for example, take the Enable thing. If it's not defined then it defaults to `false`, right? Well if we group them, than the group will be a pointer of course. Which makes enable obsolete because we have to do a nil check for the group in any case. So the enable check then becomes a check if the group is not nil which can look something like this:

```go
if r.Spec.NetworkSpec.VPC.IPv6 != nil {
```

instead of

```go
if r.Spec.NetworkSpec.VPC.EnableIPv6 {
```

Which might not look good in some cases? ¯\\\_(ツ)\_/¯. What do you think @richardcase?

Reply:
Thats a good point, the fact that the `IPv6` is defined and not nil does indicate that the user wants a ipv6 cluster. Sometimes we provide a helper function on the scope to make the intent clearer. So something like:

```go
func (s *scope) IsIPV6Enabled() bool {
   return s.ControlPlane.Spec.NetworkSpec.VPC.IPv6 != nil
}
```

If we did want an explicit enable then you can always use the kubebuilder default comment. But as you point out its not needed if we group the options.

Reply:
Good point! :) Will do that.

Reply:
Ahh, we need the enable switch because you can't do this:
```
    vpc:
      ipv6:
```

Since the rest of the fields under ipv6 are all optionals. :smile: :smile: :smile: So I need something that the user puts in. But I think most of the times this will look like this:
```
    vpc:
      ipv6:
        enable: true
```

Because I don't think people use their own IPv6 pool that often. But I guess it's fine. :slightly_smiling_face: It looks a bit weird instead of:

```yaml
   vpc:
     enableIPv6: true
```

I'll run a poll :D

Reply:
Actually, it can be something like:
```yaml
vpc:
  ipv6: {}
```
Which is fine, that's how GitHub actions work as well.

Reply:
Done :)